Migrate: Task Submission Card

KeenanRC opened the pull request.

mslavnenko committed:
NEW: Add tab to upload task assessment resources

mslavnenko committed:
Merge pull request #1 from mslavnenko/assessment_resources_tab

NEW: Add tab to upload task assessment resources

mslavnenko committed:
NEW: Add task assessment comment and modal.

mslavnenko committed:
NEW: Add Routing Key input and Task Assessment checkbox

Enable automated task assessment and pass the language per unit and task

mslavnenko committed:
FIX: Fix a small typo.

mslavnenko committed:
LOOKS: Change 'Task Assessment' to 'Overseer Assessment' labels in unit and task settings

mslavnenko committed:
NEW: Add assessment results comms from API to UI and map it to the assessment comment

mslavnenko committed:
LOOKS: Modify task assessment comment look and change date when submission is processed.

mslavnenko committed:
LOOKS: Make assessment comment a card that displays an assessment summary.

mslavnenko committed:
LOOKS: Add date of task assessment comment.

mslavnenko committed:
ENHANCE: Add method to task submission service to get latest submission timestamps

mslavnenko committed:
LOOK: Add submission history component that is now used in the task assessment comment modal.

I will also be used on a separate page with submission history.

mslavnenko committed:
NEW: Add Docker Image Name dropdown to unit and task definition editors.

mslavnenko committed:
FIX:  use updated task object after submission if Overseer is enabled.

mslavnenko committed:
LOOKS:  make Submission-history component Material Design instead of Ant Design.

Styling is not finished.

mslavnenko committed:
LOOKS: make task assessment modal responsive

mslavnenko committed:
QUALITY: remove unnecessary CSS for task submission history component.

mslavnenko committed:
LOOKS: Add test submission history to admin UI for unit chair in task definition editor.

mslavnenko committed:
FIX: fix linting error in coffeescript that prevented doubtfire-web from starting.

mslavnenko committed:
FIX: remove Ant Design imports from Angular module.

mslavnenko committed:
FIX: remove And Design from package.json and styles.

mslavnenko committed:
FIX:  remove Ant Design import from task assessment modal.

mslavnenko committed:
LOOKS: Add tabs to task submission history component.

Connect history component to API.

mslavnenko committed:
LOOKS: Refresh button added to the submission history component in case no submissions returned from the API.

mslavnenko committed:
FIX: use unit_id only instead of passing the whole unit to task on test submission.

Passing the whole unit cause an API error when saving a task definition.

mslavnenko committed:
ENHANCE: use unit chair's project_id to make test submissions.

mslavnenko committed:
LOOKS: hide all submission history and assessment components if Overseer is disabled in the app settings.

mslavnenko committed:
QUALITY: make IsOverseerEnabled flag a BehaviourSubject of boolean instead of just a boolean

mslavnenko committed:
QUALITY: separate method to check whether there are any submissions.

mslavnenko committed:
FIX: make IsOverseerEnabled a regular boolean instead of BehaviorSubject.

mslavnenko committed:
QUALITY: Refactor IsOverseerEnabled flag usage.

mslavnenko committed:
FIX: hide task assessment tab in admin UI if the overseer is disabled.

mslavnenko committed:
ENHANCE: change timestamp array to an array of objects with additonal data.

mslavnenko committed:
FIX: hide the error popup when the latest submission isn't processed yet.

mslavnenko committed:
FIX: open previously selected tab when Refresh button is clicked.

mslavnenko committed:
LOOKS: Add task status icon to submission history tab in submissioh history component.

mslavnenko committed:
FIX: add input binding to taskStatus to task status icon template.

mslavnenko committed:
LOOKS: show task status icons at the right.

KeenanRC committed:
Task Submission card - Recover old works

macite committed:
FIX: Add missing create portfolio task in inbox

macite committed:
LOOKS: Remove excess padding from create portfolio task

macite committed:
ENHANCE: Add ability to search by extension in inbox

jakerenzella committed:
FIX: Change how markedown pipe marks the text

jakerenzella committed:
QUALITY: Update angularjs dependencies

jakerenzella committed:
Angular CLI update for packages - @angular/core@latest, @angular/cli@latest

jakerenzella committed:
QUALITY: Update flex-layout dependency

jakerenzella committed:
Angular CLI update for packages - @angular/material@latest

jakerenzella committed:
QUALITY: Update packages

jakerenzella committed:
Angular CLI update for packages - rxjs@latest

macite committed:
ENHANCE: Switch do not resubmit to feedback exceeded

macite committed:
FIX: Remove unused convenors value from unit

macite committed:
FIX: Use switch to tutorial in "project" object

macite committed:
ENHANCE: Hide enrol in tutorial when at capacity

jakerenzella committed:
QUALITY: Update and rearrange packages

jakerenzella committed:
QUALITY: Update marked package

jakerenzella committed:
QUALITY: Update dependencies

jakerenzella committed:
NEW: Add update checking service

jakerenzella committed:
NEW: Add snack bar module

jakerenzella committed:
NEW: Downgrade update service

jakerenzella committed:
NEW: Add button to force check for updates

KeenanRC committed:
Added Task-submission-component and removed TAskAPI

jakerenzella committed:
LOOKS: Fix inbox truncation

erickao5858 commented on src/app/ajs-upgraded-providers.ts:
```diff
@@ -23,6 +23,7 @@ export const commentsModal = new InjectionToken('CommentsModal');
 export const taskDefinition = new InjectionToken('TaskDefinition');
 export const groupService = new InjectionToken('groupService');
 export const plagiarismReportModal = new InjectionToken('PlagiarismReportModal');
+export const taskFeedbackService  = new InjectionToken('TaskFeedbackService ');
```
The string is not correct.

erickao5858 commented on src/app/doubtfire-angularjs.module.ts:
```diff
@@ -277,6 +276,7 @@ import { EmojiService } from './common/services/emoji.service';
 import { TaskListItemComponent } from './projects/states/dashboard/directives/student-task-list/task-list-item/task-list-item.component';
 import { CreatePortfolioTaskListItemComponent } from './projects/states/dashboard/directives/student-task-list/create-portfolio-task-list-item/create-portfolio-task-list-item.component';
 import { TaskDescriptionCardComponent } from './projects/states/dashboard/directives/task-dashboard/directives/task-description-card/task-description-card.component';
+import { TaskSubmissionnCardComponent } from './projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.component';
```
Test

KeenanRC committed:
updated ajs-upgraded-proveiders.ts

erickao5858 reviewed (commented):
Good

jakerenzella committed:
FIX: Add index.html to start_url

jakerenzella committed:
Merge pull request #355 from jakerenzella/looks/fix-inbox-truncation

LOOKS: Fix inbox truncation

jakerenzella committed:
Merge pull request #349 from macite/enhance/switch-to-feedback-exceeded

ENHANCE: Switch do not resubmit to feedback exceeded

jakerenzella committed:
Merge pull request #350 from macite/fix/minor-updates

Fix/minor updates

jakerenzella committed:
Merge pull request #352 from jakerenzella/fix/fix-markdown-rendering

Fix/fix markdown rendering

jakerenzella committed:
Merge pull request #353 from jakerenzella/quality/upgrade-angular-to-11.2.9

Quality/upgrade angular to 11.2.9

jakerenzella committed:
Merge pull request #354 from jakerenzella/new/add-service-worker-forced-refresh

New: Add service worker forced refresh

jakerenzella committed:
QUALITY: Remove console logs

KeenanRC committed:
Task Submission Card - Debug

KeenanRC committed:
Style Sheet semi done on Submission Card

KeenanRC committed:
Button Rollback and Styles Corrected

KeenanRC committed:
Button Rollback and styles corrected

KeenanRC committed:
Button Rollback and styles corrected

jakerenzella committed:
QUALITY: Update angular-eslint

jakerenzella committed:
NEW: Temporarily remove eslint

jakerenzella committed:
Angular CLI update for packages - @angular/core@12, @angular/cli@12

jakerenzella committed:
@angular/cli migration - update-angular-config-v12

Remove deprecated options from 'angular.json' that are no longer present in v12.

jakerenzella committed:
@angular/cli migration - update-zonejs

Update 'zone.js' to version 0.11.x. Read more about this here: https://github.com/angular/angular/blob/master/packages/zone.js/CHANGELOG.md#breaking-changes-since-zonejs-v0111

jakerenzella committed:
Angular CLI update for packages - @angular/material@latest

jakerenzella committed:
@angular/material migration - migration-v12

Updates Angular Material to v12

jakerenzella committed:
NEW: Add angular-eslint back

jakerenzella committed:
FIX: Update resolve with to use null paramter

jakerenzella committed:
QUALITY: Remove node-sass

jakerenzella committed:
FIX: Remove ~ reference in sass use statement

jakerenzella committed:
LOOKS: Fix padding of task comments

jakerenzella committed:
LOOKS: Fix task inbox truncation

jakerenzella committed:
Merge pull request #362 from jakerenzella/new/upgrade-angular-12

New: Upgrade angular 12

macite committed:
CONFIG: Switch deploy location

macite committed:
CONFIG: Switch api url to use origin

jakerenzella committed:
Angular CLI update for packages - @angular/core@12, @angular/cli@12

jakerenzella committed:
Angular CLI update for packages - @angular/material@latest

jakerenzella committed:
Angular CLI update for packages - @angular/flex-layout@latest

jakerenzella committed:
Merge pull request #365 from jakerenzella/development

QUALITY: Minor Angular version upgrade

macite committed:
CONFIG: Update build script to indicate memory usage

macite committed:
CONFIG: Greatly simplify the docker config

macite committed:
CONFIG: Correct development URL to use port 3000 for api

macite committed:
Merge branch 'development' of https://github.com/doubtfire-lms/doubtfire-web into development

jakerenzella committed:
chore(release): 4.0.0

jakerenzella committed:
build: add new husky and conventional-commits packages

jakerenzella committed:
chore(release): 4.0.0

jakerenzella committed:
chore(release): 4.0.1

jakerenzella committed:
chore(release): 4.0.2

jakerenzella committed:
chore(release): 4.0.0

jakerenzella committed:
build: bump to version 5

jakerenzella committed:
build: remove old .git-hooks submodule directory

jakerenzella committed:
build: remove deprecated flags from build

jakerenzella committed:
build: Migrate to ESLint

jakerenzella committed:
build: delete unused tslint.json

jakerenzella committed:
quality: migrate to eslint

jakerenzella committed:
Merge branch 'quality/add-commit-configuration' of https://github.com/doubtfire-lms/doubtfire-web into build/update-build-steps

jakerenzella committed:
refactor: adhere to new ESLint rules

jakerenzella committed:
docs: add contribution commit rules to the contribution document

macite committed:
CONFIG: Switch to node 14 for angular

macite committed:
Merge pull request #368 from jakerenzella/build/update-build-steps

Build: Update build steps

macite committed:
CONFIG: Limit actions to node 14

macite committed:
Merge branch 'development' of https://github.com/doubtfire-lms/doubtfire-web into HEAD

macite committed:
FIX: Ensure comments on submission pass correct data

jakerenzella committed:
fix(service-worker): delay sw registration and update

jakerenzella committed:
fix(service-worker): let app stabalise before checking for update

jakerenzella committed:
Merge pull request #369 from jakerenzella/fix/fix-sw-registration

Fix/fix sw registration

jakerenzella committed:
fix: disable chaching in staff task list

jakerenzella committed:
Merge pull request #370 from jakerenzella/fix/task-inbox-icons

fix: disable chaching in staff task list

jakerenzella committed:
refactor: upgrade to angular 12.0.4

jakerenzella committed:
Angular CLI update for packages - @angular/material@latest

jakerenzella committed:
Angular CLI update for packages - @angular-eslint/schematics@latest

jakerenzella committed:
build: add dependabot configuration

jakerenzella committed:
refactor: update emoji picker component

jakerenzella committed:
docs: update the migration progress in README

jakerenzella committed:
fix: remove download all portfolio button

jakerenzella committed:
Merge pull request #377 from jakerenzella/upgrade/angular-12.0.4

Upgrade/angular 12.0.4

jakerenzella committed:
refactor: remove empty final line from staff task list

jakerenzella committed:
refactor: html style changes in staff-task-list

jakerenzella committed:
feat: add sorting ui to staff task list

jakerenzella committed:
style: remove right padding from filter panel in staff task list

jakerenzella committed:
feat: add sorting state data to staff task list

jakerenzella committed:
refactor: fix typo in staff task list ts

jakerenzella committed:
feat: add disabled sort button to tutorial in staff task list

jakerenzella committed:
feat: store the original filtered list to staff task list component

jakerenzella committed:
refactor: adhere to eslint in staff task list

jakerenzella committed:
refactor: start using df prefix for all app components

jakerenzella committed:
feat: add task def sorting

jakerenzella committed:
feat: add tutorial def sorting stub

jakerenzella committed:
perf: only copy original task list on refresh not filters

jakerenzella committed:
fix: don't assign the filtered tasks by reference

jakerenzella committed:
fix: ensure sorting works when filters are applied

jakerenzella committed:
style: disable sorting when filtering by task definition

jakerenzella committed:
refactor: remove debug output

macite committed:
Merge branch 'assessment_admin_ui' of https://github.com/doubtfire-overseer/doubtfire-web into doubtfire-overseer-assessment_admin_ui

macite committed:
fix: remove some issues from angular module

jakerenzella committed:
Merge pull request #380 from jakerenzella/enhance/add-task-inbox-sorting

Enhance/add task inbox sorting

macite committed:
chore: move institution settings and incorporate overseer

macite committed:
chore: move activity type and campus list

macite committed:
feat: incorporate updated overseer ui

macite committed:
fix: remove time format on tutorial due to required issue

macite committed:
fix: remove duplicate status icon component

macite committed:
fix: correct typo in student name pipe

macite committed:
style: move assessment comment out of bubble

macite committed:
style: simplify layout of assessment comment

macite committed:
refactor: change ready to mark to ready for feedback

macite committed:
style: adjust appearance of the assessment comment

Roboromeo1 committed:
feat(context): eslint fix

Roboromeo1 committed:
feat(context): eslint errors fixed

macite committed:
chore(release): 6.0.0

macite committed:
Merge branch 'development' into doubtfire-overseer-assessment_admin_ui

Roboromeo1 committed:
feat(quality): eslint errors fixed to warnings

Roboromeo1 committed:
fix: change eslint errors to warnings

macite committed:
Merge pull request #390 from Roboromeo1/development

config: switch eslint errors to warning to suppress CI failures

jakerenzella committed:
Angular CLI update for packages - @angular/core@12, @angular/cli@12

jakerenzella committed:
Angular CLI update for packages - @angular/material@latest

jakerenzella committed:
Angular CLI update for packages - @angular-eslint/schematics@latest

jakerenzella committed:
Merge pull request #393 from jakerenzella/quality/upgrade-angular-v12.1

Upgrade to angular v12.1

macite committed:
Merge branch 'development' into 5.0.x

macite committed:
chore: fix package version number and change log

Co-authored-by: Jake Renzella <j@kerenzella.com>

macite committed:
fix: fix incorrect comparison on submission history

macite committed:
chore: correct version in package file

macite committed:
chore(release): 5.0.0-0

macite committed:
chore(release): 5.0.0-1

macite committed:
chore(release): 5.0.0-2

macite committed:
Merge branch '5.0.x' of https://github.com/doubtfire-lms/doubtfire-web into 5.0.x

macite committed:
Merge branch '5.0.x' into migrate/task-submission-card
